### PR TITLE
docs: refine torch_api.communication API documentations

### PR DESCRIPTION
**Pull request by shjwudp**

### CODE REVIEW

File: bagua/torch_api/communication.py [hunk 1]
```diff
@@ -38,46 +38,61 @@ def is_initialized():
     return _global_state is not None
 
 
-def init_process_group(init_method: str = "dist://", device_id=None):
+def init_process_group():
+    """Initializes the default distributed process group, and this will also
+    initialize the distributed package, should be executed before all the APIs
```
Comment (NOBLES5E) [inline bagua/torch_api/communication.py:43]: the PyTorch builtin distributed process group?

File: docs/conf.py [hunk 2]
```diff
@@ -131,11 +131,24 @@
...
+    "bagua.torch_api.communication.get_bagua_hyperparameters",
+    "bagua.torch_api.communication.get_hyperparameters_service_client",
+    "bagua.torch_api.communication.gen_nccl_unique_id",
+    "bagua.torch_api.communication.init_bagua_inter_communicator",
+    "bagua.torch_api.communication.init_bagua_intra_communicator",
+    "bagua.torch_api.communication.init_bagua_communicator",
+    "bagua.torch_api.communication.broadcast_coalesced",
+    "bagua.torch_api.communication.broadcast",
+    "bagua.torch_api.communication.allreduce_coalesced",
+    "bagua.torch_api.communication.allreduce",
```
Comment (NOBLES5E) [inline docs/conf.py:145]: seem that `allreduce`, `broadcast` etc. should be available for users to call?

**Review by NOBLES5E — CHANGES_REQUESTED** [review]
see comments

File: docs/conf.py (hunk 2 above)
Comment (shjwudp) [reply docs/conf.py]: DONE, add allreduce and broadcast docs.

File: bagua/torch_api/communication.py (hunk 1 above)
Comment (shjwudp) [reply bagua/torch_api/communication.py]: DONE

**Review by NOBLES5E — CHANGES_REQUESTED** [review]
need to pass CI

**Comment by NOBLES5E** [comment]
@shjwudp ship. looks much better now!

**Pull request merged**